releases/stable-branches-and-release-cycle.md

```diff
@@ -0,0 +1,79 @@
+# Producing and supporting stable releases
+
+## Summary
+
+Today the Submariner project releases one minor version a month without capacity
+for stable branches or bugfix releases. In the following proposal we define a plan
+to make this possible starting with 0.8.
+
+## Proposal
+
+### Alignment of branch -> image releases
+
+We produce currently `<image>:devel` for every master push, along with `<image>:x.y.z` for
+the releases.
+
+Changing the `master` branch name to `devel` would make this homogeneous so our image releases
+are simply `<image>:<branch-or-tag>`.
+
+### Creating one stable branch for every minor release
+
+Approaching the release of every minor or major version a new stable branch will be forked
+off `devel` branch in the form of `release-x.y` for every existing repository.
+
+Any changes related to pinning shipyard base image will be done in that branch.
+
+At that point in time, any patches which belong to the release will need to be merged in master
+and backported to `release-x.y`. For this reason, the later we branch, the less extra work
```

Comment:
Maybe we can create the branch on release-x.y-rc0

Reply:
I rather like the idea of preparing the release in the branch, and creating the branch in preparation for rc0. That way the devel branch is always pinned to devel Shipyard in particular (we’d still need to bump dependencies in `go.mod`).

Reply:
It's a compromised between how many backports/cherry-pics will need to do, and how clean ' devel' branch stays

Reply:
We should create the branch whenever we start the release so rc0 makes sense, but as presented we'd have milestones before that which will pre-release without creating a branch for them.